models/helper_directory.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"github.com/Unknwon/com"
+)
+
+// LocalCopyPath returns the local repository temporary copy path.
+func LocalCopyPath() string {
+	if filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
+		return setting.Repository.Local.LocalCopyPath
+	}
+	return path.Join(setting.AppDataPath, setting.Repository.Local.LocalCopyPath)
+}
+
+// CreateTemporaryPath creates a temporary path
+func CreateTemporaryPath(prefix string) (string, error) {
```

Comment:
Is this compatible with existing temporary path and check out?

Reply:
Yes. It's the same code but with a different prefix - I've moved the temp repo code to use it.

What really should be doing is removing temporary repo and moving that functionality to modules/git.

The main issue preventing that is the DiffIndex code.

Reply:
Temp repo is now mostly a shim around these modules/git functions